### DISCUSSION TITLE
Error when using model created from datasets using roboflow and model training with ultralytics

### DISCUSSION THREAD

**Discussion by Leonn714** [PUTvision/qgis-plugin-deepness#190] [General]
Hello, I am new to QGIS and Deepness.  I came across this solution while researching for a solution to train AI segmentation detection.  I am looking to train my own model to detect swimming pools.  I have very little understanding of python and thus opted to follow the no code guide provided in the tutorial section.  I am able to export from QGIS with the following settings. 

![image](https://github.com/user-attachments/assets/4b578b75-cb1f-4496-a5b6-a6b8b7a18fc2)

The images exported perfectly.  Training in Roboflow was fine and I made sure to export the dataset as TXT - YOLOv8.  I traided the dataset in Ultralytics and converted the result to ONNX.  Up to this point everything is running fine.  But when I import the model into QGIS I get the following error.

![image](https://github.com/user-attachments/assets/cb217567-e3c1-4172-ba24-f4195e391815)

I have tried doing it to a smaller sample set, tried exporting it to YOLOv5, but i keep getting this error.  Can someone please point me into the right direction on what I am doing wrong?  Very much appreciated!

L

**Comment by bartoszptak** [comment]
It looks like you are loading the YOLO model as a Segmentor.

**Comment by Leonn714** [comment]
I changed my ONNX model to detector, and it shows up fine.  But does not work in the Segmentor model.  Do I need to export the dataset to something else if i want to train a segmentor model?  I would ideally like to have the detection to trace around an object rather than a box.

It looks like this in Detector mode. but still fails in segmentor mode.  The error makes it look like it is expecting some kind of mask.  If i am understanding your comment, YOLO can not be trained as segmentor model?  Sorry for the newbie question.

![image](https://github.com/user-attachments/assets/efa2a403-2771-46ae-961c-141a9b820926)
